module/project/templates/helper.js

```diff
@@ -957,74 +958,190 @@ module.exports = class ProjectTemplatesHelper {
+                    return resolve({
+                        success : false,
+                        data : result.data,
+                        message : CONSTANTS.apiResponses.PROJECT_TEMPLATE_DETAILS_FETCHED
+                    });
                 }
 
-                let result = await _templateInformation(templateData[0])
-
-                if( !result.success ) {
-                    return resolve(result);
-                }
+                //get data when link is given
+                if(link){
+                    
+                    let queryData={};
+                    queryData["link"]=link;
+
+                    let solutionDocument = 
+                    await coreService.solutionsDocuments(
+                        queryData,
+                        [
+                            "_id",
+                            "name",
+                            "programId",
+                            "programName",
+                            "externalId",
+                            "projectTemplateId",
+                            "link"
+                        ],
+                        "none"
+                        );
 
-                if( !templateData[0].isReusable ) {
+                    if( !solutionDocument.success || solutionDocument.data.length===0 ) {
+                        throw {
+                            message : CONSTANTS.apiResponses.SOLUTION_NOT_FOUND,
+                            status : HTTP_STATUS_CODE['bad_request'].status
+                        }
+                    }
                     
-                    templateData[0].projectId = "";
+                    let solutiondata=solutionDocument.data;
+                    let templateId=solutiondata.map(function(id){
+                        return id.projectTemplateId
+                    });
+                    templateId=templateId.toString();
+                    if(!templateId){
+                        // throw {
+                        //     message : CONSTANTS.apiResponses.MISSING_PROJECT_TEMPLATE_ID,
+                        //     status : HTTP_STATUS_CODE['bad_request'].status,
+                        //     result:"checking"
+                        // }
+                        return resolve({
+                            success : false,
+                            data : solutiondata,
+                            message : CONSTANTS.apiResponses.MISSING_PROJECT_TEMPLATE_ID
+                        });   
+                    }
+                    //for templateId: is available
+                    let findQuery = {};
+
+                    let validateTemplateId = UTILS.isValidMongoId(templateId);
+                    if( validateTemplateId ) {
+                        findQuery["_id"] = templateId;
+                    }else {
+                        findQuery["externalId"] = templateId;
+                    }
+                    let templateData = await projectTemplateQueries.templateDocument(findQuery,"all",
```

Comment:
@VISHNUDAS-tunerlabs remove duplicate code make it common for link and templateId